rt/rs/extensions/reactor/src/main/java/org/apache/cxf/jaxrs/reactor/client/FluxRxInvokerImpl.java

```diff
@@ -0,0 +1,178 @@
+        return method(HttpMethod.PUT, responseType);
+    }
+
+    @Override
+    public <R> Flux<R> put(Entity<?> entity, GenericType<R> genericType) {
+        return method(HttpMethod.PUT, entity, genericType);
+    }
+
+    @Override
+    public Flux<Response> post(Entity<?> entity) {
+        return method(HttpMethod.POST, entity);
+    }
+
+    @Override
+    public <R> Flux<R> post(Entity<?> entity, Class<R> responseType) {
+        return method(HttpMethod.POST, entity, responseType);
+    }
+
+    @Override
+    public <R> Flux<R> post(Entity<?> entity, GenericType<R> genericType) {
+        return method(HttpMethod.POST, entity, genericType);
+    }
+
+    @Override
+    public Flux<Response> delete() {
+        return method(HttpMethod.DELETE);
+    }
+
+    @Override
+    public <R> Flux<R> delete(Class<R> responseType) {
+        return method(HttpMethod.DELETE, responseType);
+    }
+
+    @Override
+    public <R> Flux<R> delete(GenericType<R> genericType) {
+        return method(HttpMethod.DELETE, genericType);
+    }
+
+    @Override
+    public Flux<Response> head() {
+        return method(HttpMethod.HEAD);
+    }
+
+    @Override
+    public Flux<Response> options() {
+        return method(HttpMethod.OPTIONS);
+    }
+
+    @Override
+    public <R> Flux<R> options(Class<R> responseType) {
+        return method(HttpMethod.OPTIONS, responseType);
+    }
+
+    @Override
+    public <R> Flux<R> options(GenericType<R> genericType) {
+        return method(HttpMethod.OPTIONS, genericType);
+    }
+
+    @Override
+    public Flux<Response> trace() {
+        return method(TRACE);
+    }
+
+    @Override
+    public <R> Flux<R> trace(Class<R> responseType) {
+        return method(TRACE, responseType);
+    }
+
+    @Override
+    public <R> Flux<R> trace(GenericType<R> genericType) {
+        return method(TRACE, genericType);
+    }
+
+    @Override
+    public Flux<Response> method(String name) {
+        return method(name, Response.class);
+    }
+
+    @Override
+    public <R> Flux<R> method(String name, Class<R> responseType) {
+        return flux(webClient.async().method(name, responseType));
+    }
+
+    @Override
+    public <R> Flux<R> method(String name, GenericType<R> genericType) {
+        return flux(webClient.async().method(name, genericType));
+    }
+
+    @Override
+    public Flux<Response> method(String name, Entity<?> entity) {
+        return method(name, entity, Response.class);
+    }
+
+    @Override
+    public <R> Flux<R> method(String name, Entity<?> entity, Class<R> responseType) {
+        return flux(webClient.async().method(name, entity, responseType));
+    }
+
+    @Override
+    public <R> Flux<R> method(String name, Entity<?> entity, GenericType<R> genericType) {
+        return flux(webClient.async().method(name, entity, genericType));
+    }
+
+    private <R> Flux<R> flux(Future<R> future) {
+        Flux<R> flux = Flux.from(Mono.fromFuture(toCompletableFuture(future)));
```

Comment:
This part concerns. I would suggest to not use `scheduler` but pass `executor` directly to `toCompletableFuture` (so the `supplyAsync` call would be done outside of common `ForkJoin` pool. Otherwise we have too many pools and threads involved for no real reason. 

Regarding the desired behavior, we would certainly want the subscription to be invoked on the separate thread (this is what `subscribeOn` does now). However, because we create `Flux` from `CompletableFuture`, intuitively the completion callback would be invoked from another thread, either the one from common `ForkJoin` pool or the provided one (unless then value is available immediately). 

Thoughts?

Reply:
Ok, the executor part is fixed.  If there's an executor passed in we use that.

However, there's an interesting conundrum.  I'm not sure how `rx()` is meant to work this way.  For a `Mono` object its pretty clear, I had to tweak the test to use a version of the json processor that didn't treat the result as an array.  Which makes sense, its meant to be a single object.

However, I need to figure out a way to check if the opening character when reading the result is a single or multiple result.  Effectively I need the `MessageBodyReader` equivalent of `StreamingResponseProvider`.  I'm kind of wondering how this works for the RxJava versions (but then again, there's no tests for this method, so I wonder if you guys ran into that issue already).

Reply:
On the client side one works against a specific URI and thus the expectations are known in advance, i.e, it is the collection or not which is returned

Reply:
That's what I'm thinking as well, to make it less complicated.  So what that means is if you work with it, by default you're getting `Mono` objects back (since it's typically one element).  Any method that supports a `GenericType` I'm going to add a `Flux` version for that just takes the singular attribute, so that you end up with something like `<R> Flux<R> getFlux(Class<R> responseType)` which should give a pretty clean programming model.

Reply:
Can you clarify a bit, may be with the code examples in the JIRA issue ? I'm not sure what is a distinction between a single and multiple objects on the client side, example, on the server one can be streaming a sequence of Book objects, but on the client it can be a single Books object.